### PR TITLE
Add tests to CI

### LINKED ISSUE #436: Tests in CI
#### Description of requested feature:
Since we now have tests, it would be nice to automatically run then on push and on pull requests

#### Reasons for adding feature:
To avoid breaking code

#### Extra information, such as Shutter version, operating system and ideas for how to implement:
Github actions perhaps

### PR DESCRIPTION
**Pull request by vadi2**

### CODE REVIEW

**Comment by Photon89** [comment]
I updated the required dependencies according to the list in https://packages.ubuntu.com/jammy/shutter However, GitHub seems to be using Ubuntu 20.04 in CI which is missing two dependencies: libgoocanvas2-cairotypes-perl and libgtk3-imageview-perl 

We need to either wait till GitHub switches to 22.04 or install those by hand.

**Comment by vadi2** [comment]
Hmm. We could try running the tests in Docker then, but I haven't used that with actions myself.

**Comment by Photon89** [comment]
Maybe we can make it work the way it is, the only problem now is that the libs installed by cpanm are somewhere outside of @INC so perl doesn't find them. Probably they are somewhere in $HOME and this folder needs to be added to @INC. I am testing on an Ubuntu 20.04 VM and can reproduce the problem but don't yet know how to solve it...

**Comment by DarthGandalf** [comment]
Add it to inc ? Either directly or via local::lib

ср, 2 мар. 2022 г., 12:11 Michael Kogan ***@***.***>:

**Comment by vadi2** [comment]
What is inc?

**Comment by Photon89** [comment]
Sorry, guys, my bad. The "--install-deps" option does what it tells to do, only installs the deps, not the actual module. I'm on it.

**Comment by Photon89** [comment]
Well, now everything seems to be in place but some of the tests seem to fail because Shutter tries to do GUI stuff but we are in a CLI environment.

**Comment by DarthGandalf** [comment]
Use xvfb

ср, 2 мар. 2022 г., 12:38 Michael Kogan ***@***.***>:

**Comment by Photon89** [comment]
Looks like passing TEST_APP_SHUTTER_PATH=$(pwd) does not work any more when appended to the xvfb call, it should be prepended instead, but I don't see how to achieve this in the CI yaml file's syntax...

**Comment by Photon89** [comment]
Looks like, I messed up so badly that the tests don't even try to run any more. :smile: Why do we need this TEST_APP_SHUTTER_PATH variable at all and cannot we just hardcode it for the CI test?

**Comment by Photon89** [comment]
Finally! :smile:

**Comment by vadi2** [comment]
Hot! Great work finishing this out. I can't approve, but you can 😃

**Comment by Photon89** [comment]
Thanks! I think, @ruzhnikov and @DarthGandalf should look at this, as they should know, how the tests should turn out.

**Comment by DarthGandalf** [comment]
@vadi2 

> What is inc?

https://perldoc.perl.org/variables/@INC
https://metacpan.org/pod/local::lib

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          libcarp-always-perl libfile-basedir-perl libfile-copy-recursive-perl libfile-which-perl \
+          libglib-object-introspection-perl libglib-perl libgoocanvas2-perl libgoocanvas-2.0-dev \
+          libgtk3-perl libimage-exiftool-perl libimage-magick-perl \
+          libjson-maybexs-perl libjson-perl liblocale-gettext-perl liblwp-protocol-https-perl \
+          libnet-dbus-perl libnet-dbus-glib-perl libnet-oauth-perl libnumber-bytes-human-perl \
+          libpango-perl libpath-class-perl libproc-processtable-perl libproc-simple-perl \
+          libreadonly-perl librsvg2-common libsort-naturally-perl libwww-mechanize-perl libwww-perl \
+          libx11-protocol-other-perl libx11-protocol-perl libxml-simple-perl procps xdg-utils
+            
+          curl -L http://cpanmin.us | perl - --sudo App::cpanminus
```
Comment (DarthGandalf) [inline .github/workflows/run-tests.yml]: This is dangerous, please don't curl http into execution, even without any encryption. Ubuntu has `cpanminus` package

Comment (Photon89) [reply .github/workflows/run-tests.yml]: Thanks, fixed! For some reason it now fails to download two of the packages which it downloaded just fine previously...

**Comment by Photon89** [comment]
Turned out, the package database got outdated after some hours, I inserted an `apt update` so it works now.

Ready to merge again!

**Comment by Photon89** [comment]
If there are no further comments on necessary changes, I'll merge the PR tomorrow.

**Pull request merged**